### DISCUSSION TITLE
Hugo fails to generate description for article written after Sep 4

### DISCUSSION THREAD

**Discussion by TetraTheta** [hbstack/.github#76] [Issues and bugs]
I know this may look weird issue, but this actually happens.

![001](https://github.com/user-attachments/assets/adfa50ef-ab0f-4c00-85d1-adf40a6e85aa)

I checked generated HTML and there was no 'description' meta tag too.

Both article have very similar front matter, so there can't be any issue about front matter.

Both Hugo v0.134.0 and v0.134.1 can't generate description from the content of the article about left article written at Sep 6 (committed and pushed at Sep 7)

All dependencies are updated to latest, for both Node packages and Hugo Modules.

I really don't understand why articles before Sep 4 have description while an article that is just written today (or yesterday) doesn't have description.

**Comment by razonyang** [comment]
You should provide a reproducible repo.

**Reply by TetraTheta** [reply]
I'll try to make a minimum reproducible repo from [the original repo](https://github.com/TetraTheta/TetraLogSource).

I'm not sure which one to preserve for reproduction, but I'll try.

**Reply by TetraTheta** [reply]
I invited you to the minimum reproduction repository. `npm install && hugo serve -p 44` will launch test server.

**Reply by razonyang** [reply]
The cause is that `{{< gallery/image src="001|002" >}}` in the very beginning of content, then the Summary was `plainify` for making sure some content won't break layout, since Summary was changed since Hugo v0.134.0, which is in type HTML.

If you require the gallery to be displayed in the very beginning, just specify `summary` in front matter, if not, you can put some text before that gallery.

**Reply by TetraTheta** [reply]
Is there any way to configure Hugo or theme to ignore that shortcode when getting summary? Or, at least, revert to old behavior of getting summary without front matter.

I have lots of articles, and I don't want to check every article to find out if it starts with shortcode or not.

**Reply by razonyang** [reply]
I don't think there is, the theme just uses the Summary returned by Hugo.

**Reply by razonyang** [reply]
It's easy to find out which articles will produce an empty summary by using RegExp in this use case, take VS Code as an example, `[-\+]{3}\n+\{\{< gallery/image`.

1. `[-\+]{3}` represents the end (---, +++) of front matter.
2. `\n+` represents there will be at least one new line between content and front matter.

![image](https://github.com/user-attachments/assets/759f9bf5-a485-4db5-ad6b-8e757eb9b4d4)

**Reply by TetraTheta** [reply]
Since I couldn't think of what to add in front of posts that start with shortcode, I tried to directly modify the theme layout file, especially the `$summary | plainify` part.

And it turned out that Hugo returns `.Page.Summary` wrong if the page starts with either these:

* `![]()`: Standard Markdown syntax for displaying images
* `{{<>}}` or `{{%%}}`: Hugo shortcode syntax

Maybe if the truncated summary only consists of HTML elements, `.Page.Summary` becomes _blank_.

![001](https://github.com/user-attachments/assets/46762f3f-7168-45bd-8dd4-0a5d3e9c43af)

I didn't check the bottom part of my [production site](https://tetralog.haipa.xyz/) when posting this discussion, and other posts are also affected! 

So I added quick band-aid code to [`layouts/partials/hb/modules/blog/post/summary.html`](https://github.com/hbstack/blog/blob/385ed81f9170ddc7c1351839d579a4c28453b658/layouts/partials/hb/modules/blog/post/summary.html) like this.
```html
{{- $summary := .Page.Summary }}
{{- $truncated := .Page.Truncated }}
{{- with .Page.Description }}
  {{- $summary = . }}
  {{- $truncated = true }}
{{- end }}
{{- /* from here */ -}}
{{- if eq (len ($summary | plainify)) 1 -}}
  {{- /* $summary = .Page.Content | plainify | truncate (.Page.Site.Params.ParamSummaryLength) */ -}}
  {{- $summary = .Page.Content | plainify | truncate 200 -}}
  {{- /* printf "%d" (len $summary) */ -}}
{{- end -}}
{{- /* to here */ -}}
<div class="hb-blog-post-summary card-text text-secondary mb-auto">
  {{- $summary | plainify -}}
</div>
{{- if and .ReadMore $truncated }}
  <div class="mt-2">
    <a
      class="text-secondary"
      href="{{ .URL }}"
      {{ if $.External }}target="_blank" rel="external"{{ end }}>
      {{- i18n "read_more_about" .Page.Title -}}
    </a>
  </div>
{{- end }}
```

Since I always use `auto`-summary for every post, I didn't check if the summary should be printed or not, or other things. `.Page.ContentWithoutSummary` was also broken, so I had to use `.Page.Content`.

The length of the `$summary` is not consistent. Maybe that's because my site configuration is like this:

```yml
# hugo.yml
hasCJKLanguage: false # makes summary length inconsistent. set this as false.
summaryLength: 25

params:
  ParamSummaryLength: 25 # Keep this value same with 'summaryLength' in 'hugo.yml'. I can't access it in template.
```

I need to polish this and move it into hook if there is any available, but at least it works for my taste.

**Reply by razonyang** [reply]
I haven't read it, but it's fine, you're taking this at your own risk.

**Comment by razonyang** [comment]
For others, this issue should be resolved since Hugo's `v0.134.2`.
